src/coreclr/binder/inc/stringlexer.inl

```diff
@@ -71,172 +70,63 @@ void StringLexer::PushCharacter(WCHAR wcCurrentChar,
 
     m_wcCurrentChar = wcCurrentChar;
     m_fCurrentCharIsEscaped = fIsEscaped;
+    m_cursor--;
 }
 
 WCHAR StringLexer::GetRawCharacter()
 {
     WCHAR wcCurrentChar = 0;
 
-    if (m_cursor <= m_end)
+    if (m_cursor < m_end)
     {
         wcCurrentChar = m_cursor[0];
-        m_fReadRawCharacter = TRUE;
         m_cursor++;
-    }
-    else
-    {
-        m_fReadRawCharacter = FALSE;
-    }
-
-    return wcCurrentChar;
-}
-
-void StringLexer::PushRawCharacter()
-{
-    if (m_fReadRawCharacter)
-    {
-        m_cursor--;
-        m_fReadRawCharacter = FALSE;
-    }
-}
 
-WCHAR StringLexer::DecodeUTF16Character()
```

Comment:
This was only used when lexing numbers in escaped unicode characters like `\u12345` .
Since we are dropping `\u`, the whole support for surrogate pairs is dead code.